packages/grid/x-data-grid/src/models/colDef/gridColDef.ts

```diff
@@ -236,13 +241,21 @@ export interface GridActionsColDef extends GridColDef {
   getActions: (params: GridRowParams) => React.ReactElement<GridActionsCellItemProps>[];
 }
 
-export type GridEnrichedColDef = GridColDef | GridActionsColDef;
+export type GridEnrichedColDef<R extends GridValidRowModel = any, V = any, F = V> =
+  | GridColDef<R, V, F>
+  | GridActionsColDef;
 
-export type GridColumns = GridEnrichedColDef[];
+export type GridColumns<R extends GridValidRowModel = any> = GridEnrichedColDef<R>[];
 
-export type GridColTypeDef = Omit<GridColDef, 'field'> & { extendType?: GridNativeColTypes };
+export type GridColTypeDef<V = any, F = V> = Omit<GridColDef<V, any, F>, 'field'> & {
+  extendType?: GridNativeColTypes;
+};
 
-export type GridStateColDef = GridEnrichedColDef & {
+export type GridStateColDef<R extends GridValidRowModel = any, V = any, F = V> = GridEnrichedColDef<
```

Comment:
What do you think about using more descriptive naming for generic types?
For example `TRow/TRowModel` instead of `R` etc?

Reply:
@m4theushw was in favor of one letter generics.
One thing to consider is that on the doc, the generics already make the api lists heavier, it would probably become very heavy visually with longer names (see the `getCellParams` line [here](https://deploy-preview-4064--material-ui-x.netlify.app/api/data-grid/grid-api/)

Reply:
For me personally, if longer name is more descriptive (i.e. I don't have to guess what it means) - it's much better DX.
But if one letter generics is a convention we adopted and agreed on - it's fine with me.
I'm curious to hear @m4theushw thoughts for this specific case.

BTW, is it possible to use JSDoc to have description for generic types?

Reply:
> BTW, is it possible to use JSDoc to have description for generic types?

Yes
https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#template

I used it on this PR, but probably not everywhere.